universe/vault/types.proto

```diff
@@ -8,14 +8,22 @@ package foundation.universe.vault;
 
 option go_package = "namespacelabs.dev/foundation/universe/vault";
 
+message AppRole {
+    string name = 1;
+    string auth_method = 2;
+    VaultProvider provider = 3;
+}
+
 message Certificate {
-    string               base_domain = 1;
-    CertificateAuthority ca          = 2;
+    string base_domain = 1;
+    string issuer = 3;
+    VaultProvider provider = 4;
+
+    reserved 2;
 }
 
-message CertificateAuthority {
-    string vault_address   = 1;
-    string vault_namespace = 2;
-    string issuer          = 3;
-    string auth_method     = 4;
+message VaultProvider {
+    string address = 1;
+    string namespace = 2;
+    string auth_method = 4;
```

Comment:
```suggestion
    string auth_method = 3;
```
As this is a new message.

Reply:
Ah right. I renamed an old message, but updated the field number in the parent, so it is indeed a new message now.